docker/run.sh

```diff
@@ -231,20 +267,84 @@ if [ "$DOCKER_HQ_OVERLAY" == "none" ]; then
 else
     # commcare-hq source overlay prevents modifications in this container
     # from leaking to the host; allows safe overwrite of localsettings.py
-    rm -rf lib/overlay  # clear source overlay
-    mkdir -p commcare-hq lib/overlay lib/node_modules lib/staticfiles
+    rm -rf lib/overlay  # clear source overlay (if it exists)
+    mkdir -p commcare-hq lib/{overlay,node_modules,staticfiles}
     ln -s /mnt/lib/node_modules lib/overlay/node_modules
     ln -s /mnt/lib/staticfiles lib/overlay/staticfiles
     logmsg INFO "mounting $(pwd)/commcare-hq via $DOCKER_HQ_OVERLAY"
     if [ "$DOCKER_HQ_OVERLAY" == "overlayfs" ]; then
         rm -rf lib/work
         mkdir lib/work
-        mount -t overlay -olowerdir=commcare-hq-ro,upperdir=lib/overlay,workdir=lib/work overlay commcare-hq
+        overlayopts="lowerdir=/mnt/commcare-hq-ro,upperdir=/mnt/lib/overlay,workdir=/mnt/lib/work"
+        if truthy "$DOCKER_HQ_OVERLAYFS_METACOPY"; then
+            # see: https://www.kernel.org/doc/html/latest/filesystems/overlayfs.html#metadata-only-copy-up
+            # Significantly speeds up recursive chmods (<1sec when enabled
+            # compared to ~20sec when not). Provided as a configurable setting
+            # since there are security implications.
+            overlayopts="metacopy=on,${overlayopts}"
+        fi
+        mount -t overlay -o"$overlayopts" overlay commcare-hq
+        if truthy "$DOCKER_HQ_OVERLAYFS_CHMOD"; then
+            # May be required so cchq user can read files in the mounted
+            # commcare-hq volume. Provided as a configurable setting because it
+            # is an expensive operation and some container ecosystems (travis
+            # perhaps?) may not require it. I suspect this is the reason local
+            # testing was not working for many people in the past.
+            logmsg INFO -n "chmod'ing commcare-hq overlay... "
+            now=$(date +%s)
+            # add world-read (and world-x for dirs and existing-x files)
+            chmod -R o+rX commcare-hq
+            delta=$(($(date +%s) - $now))
+            echo "(delta=${delta}sec)" >&2  # append the previous log line
+        fi
     else
-        mount -t aufs -o br=lib/overlay:commcare-hq-ro none commcare-hq
+        # TODO: use overlayfs and drop support for aufs
+        mount -t aufs -o br=/mnt/lib/overlay:/mnt/commcare-hq-ro none commcare-hq
```

Comment:
I don't think anyone is using this in a dev setup, so if this is not required for Travis it's probably fine to drop it.

Reply:
I'll test this and put it in a separate PR.